src/toil/test/__init__.py

```diff
@@ -140,20 +140,41 @@ def needs_aws(test_item):
         dot_boto_path = os.path.expanduser('~/.boto')
         dot_aws_credentials_path = os.path.expanduser('~/.aws/credentials')
         hv_uuid_path = '/sys/hypervisor/uuid'
-        if (os.path.exists(dot_boto_path)
-            or os.path.exists(dot_aws_credentials_path)
-            # Assume that EC2 machines like the Jenkins slave that we run CI on will have IAM roles
-            or os.path.exists(hv_uuid_path) and file_begins_with(hv_uuid_path,'ec2')):
+        try:
+            with open(dot_boto_path) as f:
+                if 'aws_access_key_id' in f.read().decode('utf8'):
```

Comment:
My mistake, I think we are talking about different things. I was looking here 
http://boto.cloudhackers.com/en/latest/boto_config_tut.html
My understanding is that the config file is just another location to specify credentials which could also be commented out. Ill look at the config module and get back to you!